DxDispatch/src/model/ImageReaderWriter.cpp

```diff
@@ -0,0 +1,271 @@
+        default: throw std::invalid_argument("Unsupported channel order");
+    }
+
+    ComPtr<IWICImagingFactory> wicFactory;
+    THROW_IF_FAILED(CoCreateInstance(
+        CLSID_WICImagingFactory,
+        nullptr,
+        CLSCTX_INPROC_SERVER,
+        IID_PPV_ARGS(&wicFactory)
+    ));
+
+    ComPtr<IWICBitmapDecoder> bitmapDecoder;
+    THROW_IF_FAILED(wicFactory->CreateDecoderFromFilename(
+        srcPath.wstring().data(),
+        nullptr,
+        GENERIC_READ,
+        WICDecodeMetadataCacheOnLoad,
+        &bitmapDecoder
+    ));
+
+    ComPtr<IWICBitmapFrameDecode> bitmapFrame;
+    THROW_IF_FAILED(bitmapDecoder->GetFrame(0, &bitmapFrame));
+
+    ComPtr<IWICBitmapSource> bitmapSource = bitmapFrame;
+
+    UINT imageWidth, imageHeight;
+    THROW_IF_FAILED(bitmapFrame->GetSize(&imageWidth, &imageHeight));
+
+    WICPixelFormatGUID imagePixelFormat;
+    THROW_IF_FAILED(bitmapFrame->GetPixelFormat(&imagePixelFormat));
+
+    if (imagePixelFormat != desiredImagePixelFormat)
+    {
+        Microsoft::WRL::ComPtr<IWICFormatConverter> formatConverter;
+        THROW_IF_FAILED(wicFactory->CreateFormatConverter(&formatConverter));
+
+        THROW_IF_FAILED(formatConverter->Initialize(
+            bitmapFrame.Get(),
+            desiredImagePixelFormat,
+            WICBitmapDitherTypeNone,
+            nullptr,
+            0.0f,
+            WICBitmapPaletteTypeCustom
+        ));
+
+        bitmapSource = formatConverter;
+    }
+
+    if (imageWidth != dstTensorInfo.width || imageHeight != dstTensorInfo.height)
+    {
+        // Scale image to match the tensor dimensions.
+        ComPtr<IWICBitmapScaler> scaler;
+        THROW_IF_FAILED(wicFactory->CreateBitmapScaler(&scaler));
+        THROW_IF_FAILED(scaler->Initialize(
+            bitmapSource.Get(),
+            dstTensorInfo.width,
+            dstTensorInfo.height,
+            WICBitmapInterpolationModeHighQualityCubic
+        ));
+
+        bitmapSource = scaler;
+    }
+
+    // WIC CopyPixels writes to an interleaved (HWC) buffer with 8 bits (uint8) per channel element.
+    std::vector<std::byte> pixelBuffer(dstTensorInfo.height * dstTensorInfo.width * dstTensorInfo.channels);
+    WICRect pixelBufferRect = { 0, 0, static_cast<INT>(dstTensorInfo.width), static_cast<INT>(dstTensorInfo.height) };
+    const uint32_t pixelBufferStride = dstTensorInfo.width * dstTensorInfo.channels * sizeof(uint8_t);
+
+    THROW_IF_FAILED(bitmapSource->CopyPixels(
+        &pixelBufferRect,
+        pixelBufferStride,
+        static_cast<uint32_t>(pixelBuffer.size()),
+        reinterpret_cast<BYTE*>(pixelBuffer.data())
+    ));
+
+    switch (dstTensorInfo.dataType)
+    {
+        case DML_TENSOR_DATA_TYPE_FLOAT32: return ConvertPixelsToTensor<float>(pixelBuffer, dstTensorInfo);
+        case DML_TENSOR_DATA_TYPE_FLOAT16: return ConvertPixelsToTensor<half_float::half>(pixelBuffer, dstTensorInfo);
+
```

Comment:
I see `DML_TENSOR_DATA_TYPE_INT8` above in `GetDataTypeSize`. Should we include that data type here as a nop memcpy `case`?

Reply:
Actually won't necessarily be a nop since this also converts HWC to CHW, depending on layout. I'm not inclined to implement this right now in this PR, since usually we're binding FP32/FP16 data that gets quantized. Might address later.